container/build.py

```diff
@@ -259,16 +262,26 @@ def cloud_build(project,
       full_path = os.path.join(dirpath, f)
       os.chmod(full_path, 0o644)
 
+  # create the required steps for the GCB yaml file as a list of dicts
+  gcb_steps = get_gcb_steps(project, container, tag, package, target,
+    bazel_version, bucket, tarball, version_step=True, build_step=True,
+    retag_step=True, test_step=True)
+
+  # add a step to upload a tarball of debian packages if bucket was specified
+  if bucket:
+    gcb_steps.append(get_upload_debs_step(container, tag, package,
+                                          bazel_version, bucket, tarball))
+
+  # generate the whole GCB yaml file as a python dict
+  gcb_yaml_dict = create_gcb_yaml_dict(steps=gcb_steps, timeout='21600s',
+    images=['gcr.io/{}/{}:{}'.format(project, container, tag)])
+
   # Gets the yaml relative to the bazel-toolchains root, regardless of what directory it was called from
-  # MUST BE UPDATED IF THE YAML FILE IS MOVED
-  config_file = "{}/container/cloudbuild.yaml".format(bazel_toolchains_base_dir)
-  extra_substitution = ",_BUCKET={},_TARBALL={}".format(bucket, tarball)
-  if not bucket:
-    # Gets the yaml relative to the bazel-toolchains root, regardless of what directory it was called from
-    # MUST BE UPDATED IF THE YAML FILE IS MOVED
-    config_file = "{}/container/cloudbuild_no_bucket.yaml".format(
-        bazel_toolchains_base_dir)
-    extra_substitution = ""
+  # MUST BE UPDATED IF THE YAML FILE IS GENERATED ELSEWHERE
+  config_file = "{}/container/cloudbuild-{}.yaml".format(bazel_toolchains_base_dir, type_)
+
+  # create the GCB yaml file
+  gen_gcb_yaml_file(gcb_yaml_dict, config_file)
```

Comment:
Let's add some basic tests that validate that a gcb yaml file generated match some expected content.

Reply:
I added unit tests under test/container.